pkg/subctl/cmd/join.go

```diff
@@ -185,12 +189,63 @@ func joinSubmarinerCluster(config *rest.Config, subctlData *datafile.SubctlData)
 	clusterCIDR, err = getPodCIDR(clusterCIDR, networkDetails)
 	exitOnError("Error determining the pod CIDR", err)
 
+	status.Start("* Discovering multi cluster details")
+	globalNetworks := getGlobalNetworks(subctlData)
+
+	err = checkOverlappingServiceCidr(globalNetworks)
+	status.End(err == nil)
+	exitOnError("Error validating overlapping ServiceCIDRs", err)
+
+	err = checkOverlappingClusterCidr(globalNetworks)
+	status.End(err == nil)
+	exitOnError("Error validating overlapping ClusterCIDRs", err)
+
 	status.Start("Deploying Submariner")
 	err = submarinercr.Ensure(config, OperatorNamespace, populateSubmarinerSpec(subctlData))
 	status.End(err == nil)
 	exitOnError("Error deploying Submariner", err)
 }
 
+func checkOverlappingServiceCidr(networks map[string]*globalnet.GlobalNetwork) error {
+	for k, v := range networks {
+		overlap, err := globalnet.IsOverlappingCIDR(v.ServiceCIDRs, serviceCIDR)
+		if err != nil {
+			return fmt.Errorf("unable to validate overlapping ServiceCIDR: %s", err)
+		}
+		if overlap && k != clusterID {
+			return fmt.Errorf("invalid service CIDR: %s overlaps with cluster %s", serviceCIDR, k)
+		}
+	}
+	return nil
+}
+
+func checkOverlappingClusterCidr(networks map[string]*globalnet.GlobalNetwork) error {
+	for k, v := range networks {
+		overlap, err := globalnet.IsOverlappingCIDR(v.ClusterCIDRs, clusterCIDR)
```

Comment:
This API is very similar to checkOverlappingServiceCidr. Shall we have a single API and pass different args?

Reply:
Actually we don't need different args at all. I'll refactor this as single function.